spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -826,723 +826,790 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
-            withInfo(mul, s"Unsupported datatype ${left.dataType}")
-          }
-          if (decimalBeforeSpark34(left.dataType)) {
-            withInfo(mul, "Decimal support requires Spark 3.4 or later")
-          }
-          None
-
-        case div @ Divide(left, right, _)
-            if supportedDataType(left.dataType) && !decimalBeforeSpark34(left.dataType) =>
-          // Datafusion now throws an exception for dividing by zero
-          // See https://github.com/apache/arrow-datafusion/pull/6792
-          // For now, use NullIf to swap zeros with nulls.
-          val rightExpr = nullIfWhenPrimitive(right)
-
-          createMathExpression(
-            left,
-            rightExpr,
-            inputs,
-            div.dataType,
-            getFailOnError(div),
-            (builder, mathExpr) => builder.setDivide(mathExpr))
-
-        case div @ Divide(left, _, _) =>
-          if (!supportedDataType(left.dataType)) {
-            withInfo(div, s"Unsupported datatype ${left.dataType}")
-          }
-          if (decimalBeforeSpark34(left.dataType)) {
-            withInfo(div, "Decimal support requires Spark 3.4 or later")
-          }
-          None
-
-        case rem @ Remainder(left, right, _)
-            if supportedDataType(left.dataType) && !decimalBeforeSpark34(left.dataType) =>
-          val rightExpr = nullIfWhenPrimitive(right)
-
-          createMathExpression(
-            left,
-            rightExpr,
-            inputs,
-            rem.dataType,
-            getFailOnError(rem),
-            (builder, mathExpr) => builder.setRemainder(mathExpr))
-
-        case rem @ Remainder(left, _, _) =>
-          if (!supportedDataType(left.dataType)) {
-            withInfo(rem, s"Unsupported datatype ${left.dataType}")
-          }
-          if (decimalBeforeSpark34(left.dataType)) {
-            withInfo(rem, "Decimal support requires Spark 3.4 or later")
-          }
-          None
-
-        case EqualTo(left, right) =>
-          createBinaryExpr(
-            left,
-            right,
-            inputs,
-            (builder, binaryExpr) => builder.setEq(binaryExpr))
-
-        case Not(EqualTo(left, right)) =>
-          createBinaryExpr(
-            left,
-            right,
-            inputs,
-            (builder, binaryExpr) => builder.setNeq(binaryExpr))
-
-        case EqualNullSafe(left, right) =>
-          createBinaryExpr(
-            left,
-            right,
-            inputs,
-            (builder, binaryExpr) => builder.setEqNullSafe(binaryExpr))
+  /**
+   * Convert a Spark expression to protobuf.
+   *
+   * @param expr
+   *   The input expression
+   * @param inputs
+   *   The input attributes
+   * @param binding
+   *   Whether to bind the expression to the input attributes
+   * @return
+   *   The protobuf representation of the expression, or None if the expression is not supported
+   */
+  def exprToProto(
+      expr: Expression,
+      input: Seq[Attribute],
```

Comment:
Any reason to switching to singular `input` from `inputs`?
The doc above still says `inputs`. Should we switch to plural  `inputs` since it is `Seq`?

Reply:
In main, we currently have an inconsistency where `exprToProto` uses `input` and `exprToProtoInternal` uses `inputs`.

```scala
  def exprToProto(
      expr: Expression,
      input: Seq[Attribute],
      binding: Boolean = true): Option[Expr] = {
```

```scala
def exprToProtoInternal(expr: Expression, inputs: Seq[Attribute]): Option[Expr] = {
```

I changed both to `input` for consistency, but I agree that `inputs` is better. I have changed it.

Reply:
@kazuyukitanimura fyi, I fixed the above